identity-apps-core/apps/authentication-portal/src/main/webapp/smsOtp.jsp

```diff
@@ -29,15 +37,26 @@
 <%-- Branding Preferences --%>
 <jsp:directive.include file="includes/branding-preferences.jsp"/>
 
+<%!
+    private static final String LOCAL_SMS_OTP_AUTHENTICATOR_ID = "sms-otp-authenticator";
+%>
+
 <%
     request.getSession().invalidate();
 
-    // TODO: Use this once "org.wso2.identity.local.auth.smsotp.util.AuthenticatorUtils" is available to IS
-    // if (StringUtils.isBlank(tenantDomain)) {
-    //     tenantDomain = (String) session.getAttribute(IdentityManagementEndpointConstants.TENANT_DOMAIN);
-    // }
-    // int otpLength = Integer.parseInt(AuthenticatorUtils.getSmsAuthenticatorConfig("SmsOTP.OTPLength", tenantDomain));
+    if (StringUtils.isBlank(tenantDomain)) {
+        tenantDomain = (String) session.getAttribute(IdentityManagementEndpointConstants.TENANT_DOMAIN);
+    }
+
+    String authenticators = Encode.forUriComponent(request.getParameter("authenticators"));
```

Comment:
Can there be cases where authenticators param have more than one authenticator in addition to SMS OTP authenticator?

Reply:
No. I have tested the flows. When it comes to a particular authenticator page (ex: SMSOtp page), "authenticators" parameter holds only the relevant authenticator.